src/systems/systems/src/hydraulic/update_iterator.rs

```diff
@@ -0,0 +1,280 @@
+use crate::simulation::UpdateContext;
+use std::time::Duration;
+
+pub struct FixedStepLoop {
```

Comment:
**thought**: I think this type and `MaxFixedStepLoop` can really use a docs comment describing what they are. Right now to use them one has to read the code to figure out what it is they do. As these structs might be useful to other problem spaces outside of hydraulics, having better comments would be good.

Reply:
Pretty sure I had something but I might have put it in the update part not here indeed, will write something!